Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -77,28 +99,37 @@ - (CGSize)viewportSize {
 }
 
 /**
- Implementaiton of the ARSessionDelegate protocol. Called every frame during the active ARKit session.  Updates the AR Camera texture, and Camera pose.
+ Implementation of the ARSessionDelegate protocol. Called every frame during the active ARKit session.  Updates the AR Camera texture, and Camera pose.
  If a size change is detected also sets the UVs, and FoV values.
 */
 - (void)session:(ARSession *)__unused session didUpdateFrame:(ARFrame *)frame {
-    // Next update both metal textures used by the renderer to display the camera image.
-    _cameraTextureY = [self updateCameraTexture:frame.capturedImage plane:0];
-    _cameraTextureCbCr = [self updateCameraTexture:frame.capturedImage plane:1];
-     
-    // Check if our orientation or size has changed and update camera UVs if necessary.
-    if ([self checkAndUpdateCameraUVs:frame]) {
-        // If our camera UVs updated, then also update the FoV to match the updated UVs.
-        [self updateFoV:frame.camera];
+    @autoreleasepool{
+        // Update both metal textures used by the renderer to display the camera image.
+        CVMetalTextureRef newCameraTextureY = [self getCameraTexture:frame.capturedImage plane:0];
+        CVMetalTextureRef newCameraTextureCbCr = [self getCameraTexture:frame.capturedImage plane:1];
+        
+        // Swap the camera textures, do this under synchronization lock to prevent null access.
+        @synchronized(self) {
```

Comment:
Please review this usage, and let me know if you think this will be okay or if we should find another way to handle the lifetime of these textures either by moving renderer onto the main UI thread or find a way to properly ref count across the threads.  From what I have read p_thread_mutex_t is slightly more efficient, but once exception handling etc. has been applied the difference is negligible.  

Since we're not using [ARC](https://en.wikipedia.org/wiki/Automatic_Reference_Counting) we need to manually manage the lifetime of camera textures, but since the renderer and the ARKit session run on two separate threads we can't just blindly have the lifetime of the texture be owned by any one thread, so I wanted the render loop manually grab the current frame's texture in a threadsafe manner when it needs it then own freeing it, and the session needs to own freeing unused buffers in the cache as new frames come in from the camera.